pkg/csi_driver/reconciler.go

```diff
@@ -165,14 +174,343 @@ func (recon *multishareReconciler) reconcileWorker() {
+		shareInfoClone.Status.Error = err.Error()
+		klog.V(6).Infof("new error message found: %s, trying to update shareInfo %s", err.Error(), shareInfoClone.Name)
+		_, err := recon.updateShareInfoStatus(context.TODO(), shareInfoClone)
+		if err != nil {
+			klog.Errorf("failed to update shareInfo %s: %s", shareInfoClone.Name, err.Error())
+		}
+	}
+}
+
+func (recon *MultishareReconciler) generateNewMultishareInstance(instanceInfo *v1alpha1.InstanceInfo) (*file.MultishareInstance, error) {
+	instanceURI := util.InstanceInfoNameToInstanceURI(instanceInfo.Name)
+	project, instanceRegion, name, err := util.ParseInstanceURI(instanceURI)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse instanceURI %q: %s", instanceURI, err.Error())
+	}
+
+	network := defaultNetwork
+	connectMode := directPeering
+	kmsKeyName := ""
+
+	storageClass, err := recon.scLister.Get(instanceInfo.Spec.StorageClassName)
+	if err != nil || storageClass == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to get storageClass %q : %v", instanceInfo.Spec.StorageClassName, err)
+	}
+
+	params := storageClass.Parameters
+	for k, v := range params {
+		switch strings.ToLower(k) {
+		case paramTier:
+			if v != enterpriseTier {
+				klog.Errorf("only tier %q is supported for multishare. Ignoring %q", enterpriseTier, v)
+			}
+		case paramNetwork:
+			network = v
+		case ParamConnectMode:
+			connectMode = v
+			if connectMode != directPeering && connectMode != privateServiceAccess {
+				return nil, status.Errorf(codes.InvalidArgument, "connect mode can only be one of %q or %q", directPeering, privateServiceAccess)
+			}
+		case ParamInstanceEncryptionKmsKey:
+			kmsKeyName = v
+		case ParamReservedIPV4CIDR, ParamReservedIPRange:
+
+		case ParamMultishareInstanceScLabel, ParameterKeyLabels, ParameterKeyPVCName, ParameterKeyPVCNamespace, ParameterKeyPVName, paramMultishare:
+		case "csiprovisionersecretname", "csiprovisionersecretnamespace":
+		default:
+			klog.Errorf("Ignoring invalid parameter %q", k)
+		}
+	}
+
+	clusterLocation := recon.cloud.Zone
+	if recon.config.IsRegional {
+		var err error
+		clusterLocation, err = util.GetRegionFromZone(clusterLocation)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get region for regional cluster: %v", err.Error())
+		}
+	}
+
+	labels, err := extractInstanceLabels(params, recon.config.Name, recon.config.ClusterName, clusterLocation)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	instance := &file.MultishareInstance{
+		Project:       project,
+		Name:          name,
+		CapacityBytes: instanceInfo.Spec.CapacityBytes,
+		Location:      instanceRegion,
+		Tier:          enterpriseTier,
+		Network: file.Network{
+			Name:        network,
+			ConnectMode: connectMode,
+		},
+		KmsKeyName: kmsKeyName,
+		Labels:     labels,
+		// Description: generateInstanceDescFromEcfsDesc(m.ecfsDescription),
```

Comment:
Is this not needed?

Reply:
I think for now the code doesn't support ecfsDescription override - there's no filestore special treatment needed for this feature.

Reply:
This is something opaqueue passed through to the instance create. can you elaborate why its not supported

Reply:
fixed.